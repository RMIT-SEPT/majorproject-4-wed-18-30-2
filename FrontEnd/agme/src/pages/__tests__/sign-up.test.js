import React from "react";
import { mount } from "enzyme";
import ReactDOM from 'react-dom';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from 'react-test-renderer';



//? Pages
import SignUp from '../sign-up';

Enzyme.configure({ adapter: new Adapter() });

describe('Sign-up page test', () => {
    //? 1. Check if Sign-up Page Render Properly
    test('renders learn react link', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SignUp />, div);
    });
    //? 2. Check if there is any changes on sign-up pages
    it('Sign-up snapshot test', () => {
        const component = renderer.create(<SignUp />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    //? 3. Test the firstname form input
    it('Firstname form input', () => {
        const component = mount(<SignUp />);
        const fname = component.find("input#firstname");
        fname.instance().value = 'Aldo';
        fname.simulate('change');
        expect(fname.instance().value).toEqual("Aldo");
    })
    //? 4. Check if there is option 
    it("Show option is available in SignUp", () => {
        const component = mount(<SignUp />);
        expect(component.find('select').length).toBe(1);
    })

})
