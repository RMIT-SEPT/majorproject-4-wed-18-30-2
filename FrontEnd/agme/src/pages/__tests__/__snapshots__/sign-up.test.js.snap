// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-up page test Sign-up snapshot test 1`] = `
<form
  className="signup-form"
  data-testid="form"
  onSubmit={[Function]}
>
  <h1
    className="font-weight-bold"
    id="heading"
  >
    Sign Up
  </h1>
  <div
    className="alert alert-danger d-none"
    id="errorMessage"
  />
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="exampleSelect"
    >
      Register as
    </label>
    <select
      className="form-control"
      id="exampleSelect"
      name="select"
    >
      <option
        value="Customer"
      >
        Customer
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      First Name
    </label>
    <input
      className="form-control"
      id="firstname"
      name="firstname"
      placeholder="First Name"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Last Name
    </label>
    <input
      className="form-control"
      id="lastname"
      name="lastname"
      placeholder="Last Name"
      type="passtextword"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Email
    </label>
    <input
      className="form-control"
      id="email"
      name="email"
      placeholder="Email"
      type="email"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Password
    </label>
    <input
      className="form-control"
      id="password"
      name="password"
      placeholder="Password"
      type="password"
    />
  </div>
  <button
    aria-label={null}
    className="btn-lg btn-success btn-block mt-5 mb-3 btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    Sign Up
  </button>
  <div
    className="text-center"
  >
    <a
      href="/sign-in"
    >
      Sign up
    </a>
    <span
      className="p-2"
    >
      |
    </span>
    <a
      href="/"
    >
      Forgot Password
    </a>
  </div>
</form>
`;
